src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -81,17 +82,29 @@ public RoomDto startGame(String roomId, StartGameRequest request) {
         repository.save(room);
 
         // Initialize game state
-        createAddGameFromRoom(room);
+        Game game = createAddGameFromRoom(room);
+        setStartGameTimer(game, GameTimer.DURATION_15);
 
         RoomDto roomDto = RoomMapper.toDto(room);
         socketService.sendSocketUpdate(roomDto);
         return roomDto;
     }
 
     // Initialize and add a game object from a room object
-    public void createAddGameFromRoom(Room room) {
+    public Game createAddGameFromRoom(Room room) {
         Game game = GameMapper.fromRoom(room);
         currentGameMap.put(room.getRoomId(), game);
+        return game;
+    }
+
+    // Set and start the Game Timer.
+    public void setStartGameTimer(Game game, Long duration) {
+        GameTimer gameTimer = new GameTimer(duration);
+        game.setGameTimer(gameTimer);
+
+        // Schedule the game to end after <duration> seconds.
+        EndGameTimerTask endGameTimerTask = new EndGameTimerTask(socketService, game);
+        gameTimer.getTimer().schedule(endGameTimerTask, duration * 1000);
```

Comment:
Could this method be called from the `createAddGameFromRoom` method or absorbed into it or absorbed in the `GameMapper.fromRoom` method? I can imagine sometime in the future the duration might come from the room settings, though I guess that means we don't have to worry too much about this now

Reply:
It could be absorbed / called from either of those methods -- do you have a preference? I'm open to changes on this.

Reply:
No preference really - maybe just move the call to inside the `createAddGameFromRoom` method? Might be the easiest/quickest solution

Reply:
Sounds good -- fixed!